### PR TITLE
Refactor group create modal and conference actions

### LINKED ISSUE #1013: Double check group template initialization

### PR DESCRIPTION
**Pull request by warrenshen**
Fix #1013

### CODE REVIEW

File: app/assets/javascripts/actions/conference_actions.es6.jsx [hunk 1]
```diff
@@ -86,6 +74,14 @@
       };
     }
 
+    storeListAttribute(key, value, index) {
```
Comment (warrenshen) [inline app/assets/javascripts/actions/conference_actions.es6.jsx]: @maxwolffe how do you feel about refactoring this into the regular `storeAttribute` action?
The store would just check if an `index` is passed in to know what handle to do.

File: app/assets/javascripts/components/group/group_create_modal.es6.jsx [hunk 2]
```diff
@@ -6,11 +6,17 @@ class GroupCreateModal extends CreateModal {
...
       conference: React.PropTypes.object.isRequired,
-      groupables: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+      groupables: React.PropTypes.arrayOf(React.PropTypes.object),
       template: React.PropTypes.object.isRequired,
     };
   }
 
+  static get defaultProps() {
+    return {
+      groupables: [],
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_create_modal.es6.jsx]: This default state allows us to request `groupables` on `storeTemplate`!

File: app/assets/javascripts/components/group/group_create_modal.es6.jsx [hunk 3]
```diff
@@ -69,12 +55,25 @@ class GroupCreateModal extends CreateModal {
...
-    var attributes = this.props.template.attributes;
-    var leaders = attributes['leaderships_attributes'];
-    if (this.props.template.errors) {
-      errors = this.props.template.errors;
+  generateValue(type) {
+    var groupables = this.props.groupables;
+    var leaderships = this.props.template.attributes['leaderships_attributes'];
+    if (type === 'primary' && leaderships[0]) {
+      var id = leaderships[0]['user_id'];
+      var user = groupables.find((groupable) => groupable.id == id);
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_create_modal.es6.jsx]: Turns out there's this cool `find` function, returns after first success so no need for `pop()`.

File: app/assets/javascripts/components/group/group_create_modal.es6.jsx [hunk 4]
```diff
@@ -94,16 +93,14 @@ class GroupCreateModal extends CreateModal {
...
+            value={this.generateValue('primary')} />
           <CardDropdown
             errors={errors['leaderships_attributes']}
             choices={this.generateChoices('secondary')}
             label={'Secondary Leader'}
-            value={leaders
-                   && leaders[1]
-                   && this.findGroupable(leaders[1]['user_id']).full_name} />
+            margin={true}
+            value={this.generateValue('secondary')} />
```
Comment (maxwolffe) [inline app/assets/javascripts/components/group/group_create_modal.es6.jsx]: Nice.

File: app/assets/javascripts/components/group/group_create_modal.es6.jsx (hunk 2 above)
Comment (maxwolffe) [inline app/assets/javascripts/components/group/group_create_modal.es6.jsx]: Ah great, that makes sense.

File: app/assets/javascripts/components/conference/conference_page_overlay.es6.jsx [hunk 5]
```diff
@@ -23,13 +23,11 @@ class ConferencePageOverlay extends PageOverlay {
   // Render
   // --------------------------------------------------
   renderModal() {
+    var pairing = this.props.pairing;
     var template = this.props.template;
-    if (this.props.pairing) {
-      return <ConferenceEditModal pairing={this.props.pairing} />;
-    } else if (this.props.template.model == 'group') {
-      if (!template.attributes['leaderships_attributes']) {
-        template.attributes['leaderships_attributes'] = [null, null];
```
Comment (maxwolffe) [inline app/assets/javascripts/components/conference/conference_page_overlay.es6.jsx]: How is storeListAttribute working without this instantiated to an existing list?

File: app/assets/javascripts/actions/conference_actions.es6.jsx (hunk 1 above)
Comment (maxwolffe) [inline app/assets/javascripts/actions/conference_actions.es6.jsx]: Seems like a not unreasonable idea, but it'll require a check for an index in storeAttribute, however most storeAttributes don't require any index, and it'll make this store's `StoreAttribute` different from all other storeAttributes. I'd rather just leave it as is, but your call.

File: app/assets/javascripts/components/conference/conference_page_overlay.es6.jsx (hunk 5 above)
Comment (warrenshen) [inline app/assets/javascripts/components/conference/conference_page_overlay.es6.jsx]: It's set on template instantiation here: https://github.com/calblueprint/edge-yl/pull/1016/files#diff-196ac8815f97e973c0b9b9e9cc0d8abbR23

File: app/assets/javascripts/actions/conference_actions.es6.jsx (hunk 1 above)
Comment (warrenshen) [inline app/assets/javascripts/actions/conference_actions.es6.jsx]: You right you right. Another thing we could do is just have `primary` and `secondary` attributes in the `template`, and then on `createGroup` manually reform the params appropriately? This is an interesting case we don't see elsewhere.

File: app/assets/javascripts/components/group/group_create_modal.es6.jsx [hunk 6]
```diff
@@ -23,44 +29,24 @@ class GroupCreateModal extends CreateModal {
...
   }
 
-  generateChoices(leader) {
-    return this.props.groupables.map((groupable) => this.generateChoice(groupable, leader));
+  generateChoice(groupable, index) {
+    return {
+      action: () => ConferenceActions.storeListAttribute(
+        'leaderships_attributes',
+        { user_id: groupable.id },
+        index,
```
Comment (maxwolffe) [inline app/assets/javascripts/components/group/group_create_modal.es6.jsx]: I think you're using the wrong index here, (it should be 0 for a primary leader, and 1 for a secondary leader). I tested this, and, as is, you can't add any leader beyond choice 2.

File: app/assets/javascripts/components/conference/conference_page_overlay.es6.jsx (hunk 5 above)
Comment (maxwolffe) [inline app/assets/javascripts/components/conference/conference_page_overlay.es6.jsx]: Ahhhhh. ok ok

File: app/assets/javascripts/components/group/group_create_modal.es6.jsx (hunk 6 above)
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_create_modal.es6.jsx]: I think you're right, as I saw some funny bugs happening before I went the sleep!
I'm going to merge this in for now so we can move back to phab - we'll fix it soon!

**Pull request merged**